lxd/resources/usb.go

```diff
@@ -0,0 +1,190 @@
+package resources
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/google/gousb"
```

Comment:
Can we avoid that particular dependency and only rely on `usbid`?
That should save us from the libusb dependency.

We can't have `lxd/resources` depend on any cgo as it needs to be buildable statically and cross-compiled for use by MAAS.

Reply:
Hmm, `usbid` itself seems to be importing the main package so that may not be enough...
Will need a test compile with CGO_ENABLED=0 to see, but if the libusb requirement doesn't go away, we'll need a different approach :(

Reply:
Technically, the `github.com/google/gousb/usbid` doesn't require libusb. The whole package is just structured in such a way that there's no way around it. When moving some files around, it works just fine.

Reply:
Can we just fork and patch it?

Reply:
Yeah, having a fork of it in `shared/usbid` would probably be fine.
If there aren't too many cross-package use, it may even be something upstreamable.

Reply:
They specifically highlight the need for libusb and cgo in their README so you'd think they'd be open to not having that be a restriction on some of their sub-packages.

Reply:
I added relevant code to `shared/usbid`.